django/db/migrations/serializer.py

```diff
@@ -125,8 +125,14 @@ class EnumSerializer(BaseSerializer):
     def serialize(self):
         enum_class = self.value.__class__
         module = enum_class.__module__
+        if issubclass(enum_class, enum.Flag):
+            members, __ = enum._decompose(enum_class, self.value)
+        else:
+            members = (self.value,)
         return (
-            "%s.%s[%r]" % (module, enum_class.__qualname__, self.value.name),
+            " | ".join(
+                f"{module}.{enum_class.__qualname__}[{item.name!r}]" for item in members
+            ),
```

Comment:
A list comprehension is preferable here as `str.join()` converts to list internally anyway. It is better performance to provide a list up front.
 - https://stackoverflow.com/questions/9060653/list-comprehension-without-in-python/9061024#9061024
 - https://github.com/adamchainz/flake8-comprehensions/issues/156

```suggestion
            " | ".join([
                f"{module}.{enum_class.__qualname__}[{item.name!r}]" for item in members
            ]),
```